engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/TypeDefinitionSet.scala

```diff
@@ -0,0 +1,54 @@
+package pl.touk.nussknacker.engine
+
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import org.springframework.expression.spel.SpelNode
+import pl.touk.nussknacker.engine.api.expression.ExpressionParseError
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.definition.{DefinitionExtractor, ProcessDefinitionExtractor, TypeInfos}
+import pl.touk.nussknacker.engine.spel.TypedNode
+import pl.touk.nussknacker.engine.spel.ast.SpelAst.RichSpelNode
+
+
+class TypeDefinitionSet(typeDefinitions: Map[String, TypeInfos.ClazzDefinition]) {
+
+  def validateTypeReference(spelNode: SpelNode): Validated[NonEmptyList[ExpressionParseError], SpelNode] = {
+
+    val spelNodeChildAST = spelNode.children.headOption.getOrElse(throw new Exception("SpelNode has no children")).toStringAST
+
+    if(typeDefinitions.contains(spelNodeChildAST)) {
+      Valid(spelNode)
+    } else {
+      Invalid(NonEmptyList.of(ExpressionParseError("Class is not allowed to be passed as TypeReference")))
+    }
+  }
+
+  def displayBasicInfo: String = {
+
+    val newLine = "\n"
```

Comment:
I wonder about use: `System.lineSeparator()`? Can you check it? Try use the same thing in tab separator.

Reply:
do we need this method at all? Except for debugging/println? :)

Reply:
I deleted it, thanks